### DISCUSSION TITLE
SVG rendering problem

### DISCUSSION THREAD

**Discussion by mderouss** [asciidocfx/AsciidocFX#658] [Q&A]
I've made an SVG diagram using Inkscape to produce 'plain SVG' and saving only the part of the page that contains the actual diagram. When I include this image using the image tag ( "image::diagram.svg[]" ), then text that follows the image is located a long way down the page after the image. When I use a png image, this is not the case.  As the images I'm using are diagrams and not pictures, I'd prefer to keep them as svg if at all possible, but this behaviour will mean I have to convert them all to bitmaps. Is there a way to make text follow an svg image as it does a png image ?

**Comment by rahmanusta** [comment]
Can you please share a screenshot?

**Comment by masa300V** [comment]
@mderouss 

I'm curious to know what you have.
Are you generating PDF with FOP?
If yes, I see there are lots of limitations of our use of SVG.   :https://xmlgraphics.apache.org/fop/dev/svg.html

***
I found this interesting temporary solution.

Using a PDF to SVG converter (below link), this converter outputs very plain SVG.
I mean, I use Draw.io (or Inkscape) and output my drawings into PDF first, then use this converter.
https://github.com/dmester/pdftosvg.net/

With this converter, I use "--include-fonts none" and I use "Arial" in my drawings as I'm on Windows.

**Comment by mderouss** [comment]
Ahhhh, sorry, this now looks like finger trouble :( . Although I thought I had double checked that I was only exporting the graphic from inkscape rather than the page, when I thought to read the actual svg itself it was clear that I hadn't done that :(. When I hacked the svg to be right, the problem went away.
